### PR TITLE
AlertingNG: pause/unpause definitions via the API

### PR DESCRIPTION
**Pull request by papagian**
**What this PR does / why we need it**:
It introduces two new endpoints for pausing and unpausing an alert definition:
`/api/alert-definitions/pause`
`/api/alert-definitions/unpause`
 
Paused alert definitions aren't scheduled to be evaluated on certain intervals but they can still be evaluated using the API.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Pause example:
```
curl -H "Content-Type: application/json" -XPOST http://admin:admin@localhost:3000/api/alert-definitions/pause -d '{
    "uids": ["mh_hJLfGz","unknown"]
}'
{"message":"1 alert definitions paused"}
```

Unpause example:
```
curl -H "Content-Type: application/json" -XPOST http://admin:admin@localhost:3000/api/alert-definitions/unpause -d '{
    "uids": ["mh_hJLfGz","unknown"]
}'
{"message":"1 alert definitions unpaused"}
```

### CODE REVIEW

**Comment by kylebrandt** [comment]
Looks good to me and works. This is scope creep, but I'm wondering if we should change this to accept a list of UIDs. This way, the API would be setup for pausing multiple alert definitions in one call.

Use cases:

- Future UI: Allow for multiple alerts in the list to be selected (e.g. checkboxes) and pause/unpause them all in one call.
- Administration automation. call the list api endpoint, filter with JQ, send single call with list to pause.

File: pkg/services/ngalert/api.go [hunk 1]
```diff
@@ -180,3 +182,33 @@ func (ng *AlertNG) unpauseScheduler() response.Response {
 	}
 	return response.JSON(200, util.DynMap{"message": "alert definition scheduler unpaused"})
 }
+
+// alertDefinitionPauseEndpoint handles PUT /api/alert-definitions/pause/:alertDefinitionUID.
```
Comment (wbrowne) [inline pkg/services/ngalert/api.go:186]:
```suggestion
// alertDefinitionPauseEndpoint handles POST /api/alert-definitions/pause/:alertDefinitionUID.
```

File: pkg/services/ngalert/api.go [hunk 2]
```diff
@@ -180,3 +182,33 @@ func (ng *AlertNG) unpauseScheduler() response.Response {
...
+	}
+
+	err := ng.updateAlertDefinitionPaused(&cmd)
+	if err != nil {
+		return response.Error(500, "Failed to pause alert definition", err)
+	}
+	return response.JSON(200, util.DynMap{"message": "alert definition paused"})
+}
+
+// alertDefinitionUnpauseEndpoint handles PUT /api/alert-definitions/unpause/:alertDefinitionUID.
```
Comment (wbrowne) [inline pkg/services/ngalert/api.go:201]:
```suggestion
// alertDefinitionUnpauseEndpoint handles POST /api/alert-definitions/unpause/:alertDefinitionUID.
```

**Review by wbrowne — APPROVED** [review]
Tested and it looks good! 😄

**Comment by papagian** [comment]
> Looks good to me and works. This is scope creep, but I'm wondering if we should change this to accept a list of UIDs. This way, the API would be setup for pausing multiple alert definitions in one call.

I have changed the API calls to expect multiple definition UIDs and I have updated the PR description accordingly.

**Review by kylebrandt — APPROVED** [review]
> > Looks good to me and works. This is scope creep, but I'm wondering if we should change this to accept a list of UIDs. This way, the API would be setup for pausing multiple alert definitions in one call.
> 
> I have changed the API calls to expect multiple definition UIDs and I have updated the PR description accordingly.

Works for me. I was thinking could have the list in the url, but maybe url size issues etc, so I think I like this more.

**Comment by papagian** [comment]
Closing and re-opening in hope to get the license/cla check green.

**Pull request merged**